### ISSUE TITLE
Variables    var.py  GUI报错

### ISSUE THREAD

**Issue by ligaopan** [nokia/RED#331]
![image](https://user-images.githubusercontent.com/16896540/65207946-19a2f300-dac6-11e9-88c0-fdbdf0e9bd4d.png)
如上图，*** Settings ***下面
引用Variables    var.py
GUI上面会有报错哦
不过不影响使用哈，只是不太美观

**Comment by michalanglart** [comment]
Hi,

would it be possible to provide a description in english? Also can you show what are the errors descriptions, what is the content of `var.yaml` and `var.py` and where is `data` variable defined?

**Comment by ligaopan** [comment]
As shown above, *** settings *** below
Reference variable var.py
There will be an error on the GUI.
But it does not affect the use of Kazakhstan, but it is not very beautiful.

![image](https://user-images.githubusercontent.com/16896540/65227433-30b00800-dafb-11e9-82f0-0d275f400077.png)
![image](https://user-images.githubusercontent.com/16896540/65227465-3e658d80-dafb-11e9-841d-8f597cbb68ff.png)
${data['url']} 
Referenced is the data variable in var.py, data is the value obtained from var.yaml


My English is not very good, the above content I used google online translation, I hope to forgive me, thank you

**Comment by ligaopan** [comment]
The variable var.py line is reported as follows：

Problem importing variable file 'var.py'. Following exception has been thrown:
            robot.errors.DataError: FileNotFoundError: [Errno 2] No such file or directory: 'var.yaml'
            
            During handling of the above exception, another exception occurred:
            
            Traceback (most recent call last):
              File "D:\Python37\lib\site-packages\robot\variables\filesetter.py", line 49, in _import_if_needed
                return importer.import_variables(path_or_variables, args)
              File "D:\Python37\lib\site-packages\robot\variables\filesetter.py", line 97, in import_variables
                var_file = importer(path, instantiate_with_args=())
              File "D:\Python37\lib\site-packages\robot\utils\importer.py", line 97, in import_class_or_module_by_path
                self._raise_import_failed(path, err)
              File "D:\Python37\lib\site-packages\robot\utils\importer.py", line 109, in _raise_import_failed
                raise DataError('\n'.join(msg))
            robot.errors.DataError: Importing variable file 'D:\RED_0.8.13\yckj\workspace\ibis ui\var.py' failed: FileNotFoundError: [Errno 2] No such file or directory: 'var.yaml'
            Traceback (most recent call last):
              File "D:\RED_0.8.13\yckj\workspace\ibis ui\var.py", line 7, in <module>
                file = open(yaml_path, 'r',encoding='utf8')
            PYTHONPATH:
              C:\WINDOWS\TEMP\RobotTempDir2059546401070669730
              D:\Python37\python37.zip
              D:\Python37\DLLs
              D:\Python37\lib
              D:\Python37
              D:\Python37\lib\site-packages
            
            During handling of the above exception, another exception occurred:
            
            Traceback (most recent call last):
              File "C:\WINDOWS\TEMP\RobotTempDir2059546401070669730\robot_session_server.py", line 43, in inner
                result['result'] = func(*args, **kwargs)
              File "C:\WINDOWS\TEMP\RobotTempDir2059546401070669730\robot_session_server.py", line 105, in inner
                return to_call(*args, **kwargs)
              File "C:\WINDOWS\TEMP\RobotTempDir2059546401070669730\robot_session_server.py", line 138, in inner
                return to_call(*args, **kwargs)
              File "C:\WINDOWS\TEMP\RobotTempDir2059546401070669730\robot_session_server.py", line 78, in inner
                return func(*args, **kwargs)
              File "C:\WINDOWS\TEMP\RobotTempDir2059546401070669730\robot_session_server.py", line 193, in get_variables
                return red_variables.get_variables(path, args)
              File "C:\WINDOWS\TEMP\RobotTempDir2059546401070669730\red_variables.py", line 85, in get_variables
                vars_from_file = _get_variables_from_file(path, arguments)
              File "C:\WINDOWS\TEMP\RobotTempDir2059546401070669730\red_variables.py", line 105, in _get_variables_from_file
                variables.set_from_file(path, arguments)
              File "D:\Python37\lib\site-packages\robot\variables\variables.py", line 63, in set_from_file
                return setter.set(path_or_variables, args, overwrite)
              File "D:\Python37\lib\site-packages\robot\variables\filesetter.py", line 35, in set
                variables = self._import_if_needed(path_or_variables, args)
              File "D:\Python37\lib\site-packages\robot\variables\filesetter.py", line 53, in _import_if_needed
                % (path_or_variables, args, get_error_message()))
            robot.errors.DataError: Processing variable file 'D:\RED_0.8.13\yckj\workspace\ibis ui\var.py' failed: Importing variable file 'D:\RED_0.8.13\yckj\workspace\ibis ui\var.py' failed: FileNotFoundError: [Errno 2] No such file or directory: 'var.yaml'
            Traceback (most recent call last):
              File "D:\RED_0.8.13\yckj\workspace\ibis ui\var.py", line 7, in <module>
                file = open(yaml_path, 'r',encoding='utf8')
            PYTHONPATH:
              C:\WINDOWS\TEMP\RobotTempDir2059546401070669730
              D:\Python37\python37.zip
              D:\Python37\DLLs
              D:\Python37\lib
              D:\Python37
              D:\Python37\lib\site-packages

**Comment by ligaopan** [comment]
![image](https://user-images.githubusercontent.com/16896540/65228183-9bae0e80-dafc-11e9-990b-47c3f97dff7e.png)

**Comment by michalanglart** [comment]
We've investigated the issue. The problem you are experiencing: `[Errno 2] No such file or directory: 'var.yaml'` is caused by the relative path used by `var.py` script. This happens because RED starts a little python program inside temporary directory and it communicates with it for example to ask for variables defined in your `var.py` file. Then the script is loaded and it tries to open `var.yaml` file using relative path. However the path is relative to the temporary directory in which the server is running and the file is not there so eventually an error is reported.

What can be done:
1. We can try to change working directory when importing variable files and this would work. You would need to wait for one of next releases (next one is about to be released)

2. You can change the script so that it would use absolute directory:

```python
import yaml
import os

yaml_path = "var.yaml"
pwd_folder = os.path.dirname(os.path.abspath(__file__))
file = open (os.path.join(pwd_folder, yaml_path), "r", encoding="utf8")
file_data = file.read()
file.close()
data = yaml.load(file_data, Loader=yaml.FullLoader)
```

The line `pwd_folder = os.path.dirname(os.path.abspath(__file__))` is extracting the absolute path to the place where `var.py` file is placed. Then this path is joined with a `var.yaml` file name.

3. I'm not sure about this but are you using `yaml` import through `py` variable file because RED does not support `yaml` files? In such case we could extend RED so that it would also be able to read `yaml` variables files - after all Robot is accepting such files and it shouldn't be too hard.

**Comment by michalanglart** [comment]
Correction:

Point (3) is wrong. RED is supporting yaml files, there is no problems with doing:

```
Variables    vars.yaml
```
and the variables from this file are recognized in usage places.

**Comment by michalanglart** [comment]
Fixed in 0.9.0
